src/draw/nxp/vglite/lv_draw_vglite_line.c

```diff
@@ -55,42 +55,61 @@ static void _vglite_draw_line(const lv_point_t * point1, const lv_point_t * poin
  *   GLOBAL FUNCTIONS
  **********************/
 
-void lv_draw_vglite_line(lv_draw_task_t * t, const lv_draw_line_dsc_t * dsc)
+void lv_draw_vglite_line(vglite_draw_task_t * vglite_task)
 {
+    const lv_draw_line_dsc_t * dsc = vglite_task->t->draw_dsc;
+
     if(dsc->width == 0)
         return;
     if(dsc->opa <= (lv_opa_t)LV_OPA_MIN)
         return;
     if(dsc->p1.x == dsc->p2.x && dsc->p1.y == dsc->p2.y)
         return;
 
-    lv_layer_t * layer = t->target_layer;
+    lv_layer_t * layer = vglite_task->t->target_layer;
     lv_area_t clip_area;
     clip_area.x1 = LV_MIN(dsc->p1.x, dsc->p2.x) - dsc->width / 2;
     clip_area.x2 = LV_MAX(dsc->p1.x, dsc->p2.x) + dsc->width / 2;
     clip_area.y1 = LV_MIN(dsc->p1.y, dsc->p2.y) - dsc->width / 2;
     clip_area.y2 = LV_MAX(dsc->p1.y, dsc->p2.y) + dsc->width / 2;
 
-    if(!lv_area_intersect(&clip_area, &clip_area, &t->clip_area))
+    if(!lv_area_intersect(&clip_area, &clip_area, &vglite_task->t->clip_area))
         return; /*Fully clipped, nothing to do*/
 
     lv_area_move(&clip_area, -layer->buf_area.x1, -layer->buf_area.y1);
 
     lv_point_t point1 = {dsc->p1.x - layer->buf_area.x1, dsc->p1.y - layer->buf_area.y1};
     lv_point_t point2 = {dsc->p2.x - layer->buf_area.x1, dsc->p2.y - layer->buf_area.y1};
 
-    _vglite_draw_line(&point1, &point2, &clip_area, dsc);
+    _vglite_draw_line(vglite_task, &point1, &point2, &clip_area, dsc);
 }
 
 /**********************
  *   STATIC FUNCTIONS
  **********************/
 
-static void _vglite_draw_line(const lv_point_t * point1, const lv_point_t * point2,
+static void _vglite_set_line(int32_t * line_path, uint32_t * path_data_size, const lv_point_t * point1,
+                             const lv_point_t * point2)
+{
+    uint32_t pidx = 0;
+    line_path[pidx++] = VLC_OP_MOVE;
+    line_path[pidx++] = point1->x;
+    line_path[pidx++] = point1->y;
+    line_path[pidx++] = VLC_OP_LINE;
+    line_path[pidx++] = point2->x;
+    line_path[pidx++] = point2->y;
+    line_path[pidx] = VLC_OP_END;
+
+    *path_data_size = pidx * sizeof(int32_t);
```

Comment:
Include the last member in the size.

```suggestion
    line_path[pidx++] = VLC_OP_END;

    *path_data_size = pidx * sizeof(int32_t);
```